tools/cldr-apps/src/main/java/org/unicode/cldr/web/AdminPanel.java

```diff
@@ -0,0 +1,253 @@
+package org.unicode.cldr.web;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Random;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.unicode.cldr.util.CLDRConfig;
+import org.unicode.cldr.util.CLDRConfigImpl;
+import org.unicode.cldr.web.SurveyAjax.JSONWriter;
+
+public class AdminPanel {
+    public void getJson(JSONWriter r, HttpServletRequest request, HttpServletResponse response, SurveyMain sm) throws JSONException, IOException {
```

Comment:
should this be an abstract interface?

Reply:
Hmm i would actually go the other way.   Considering : https://github.com/unicode-org/cldr/pull/904/files#diff-00d9540e776e546a0ea8468e69245839e6f6e168af708ed3f902bf81d64ea90dR384

you could actually have a Map from string to JsonAnswerer (with a better name of course) and replace the switch statement with:

`m.get(what).getJSON(r, request, response)`

and we'd have a little dispatch table.

Reply:
Actually I was wrong: I had <em>not</em> made that change to AdminPanel, I had only thought about it. Looking at existing getJson functions --

* new AdminPanel().getJson(r, request, response, sm);
* new SurveyForumParticipation(org).getJson(r);
* new SurveyVettingParticipation(org, sm).getJson(r);
* new SurveyBulkClosePosts(sm, execute).getJson(r);
* AboutST.getJson(r, sm); 

-- AboutST is exceptional (and exceptionally short). Your JsonAnswerer idea sounds good, and might help with refactoring SurveyAjax to avoid a million "else if (what.equals(WHAT_...)) { ..." blocks. For now, I'm leaving AdminPanel as-is...